src/MyConnection.h

```diff
@@ -30,15 +35,21 @@ class MyConnection : boost::noncopyable {
     mysql_options(pConn_, MYSQL_OPT_LOCAL_INFILE, 0);
     // Default to UTF-8
     mysql_options(pConn_, MYSQL_SET_CHARSET_NAME, "UTF8");
-    if (groups != "")
-      mysql_options(pConn_, MYSQL_READ_DEFAULT_GROUP, groups.c_str());
-    if (default_file != "")
-      mysql_options(pConn_, MYSQL_READ_DEFAULT_FILE, default_file.c_str());
-
-
-    if (!mysql_real_connect(pConn_, host.c_str(), user.c_str(),
-        password.c_str(), db == "" ? NULL : db.c_str(), port,
-        unix_socket == "" ? NULL : unix_socket.c_str(), client_flag)) {
+    if ((!groups.isNull()) && (Rcpp::as<std::string>(groups) != ""))
```

Comment:
It would be nice if  `groups.get()` and not have to do all the casting, but doesn't look like Nullable supports that.

But is the check (here) and below, really necessary? What's the downside of passing on "" to `mysql_options()`?